### ISSUE TITLE
Unexpected behavior: the Silent (_) modifier does not have a cascading effect

### ISSUE THREAD

**Issue by mayrepo** [pest-parser/pest#457]
I tried to implement scheme "#;" comments (described at https://srfi.schemers.org/srfi-62/srfi-62.html) with the following line in the .pest file: 
`NODE_COMMENT = _{ "#;" ~ Node }`
`Node` is not silent itself.

But I found that `Node`s in `NODE_COMMENT`s appear in the AST because the Silent (_) modifier does not have a cascading effect.

The doc states that "Silent rules do not create token pairs during parsing, nor are they error-reported."
It seems to me that the wording of the doc does not match the actual implementation.

The behavior described in the doc seems more desirable to me (and it matches my use case) so I suggest making the Silent (_) modifier have a cascading effect rather than modifying the doc.

**Comment by mdg** [comment]
I've found that it sort of cascades. This isn't really helpful, b/c I haven't been able to decipher the exact behavior yet. But in one case, I removed the silent modifier and a bunch of inner rules appeared. What would be great is a silent operator that does cascade and a quiet or transparent operator that only affects that rule and not inner rules, similar to CompoundAtomic.

**Comment by CAD97** [comment]
Yeah, this behavior isn't entirely self-evident yet, due to the multiple jobs each modifier holds.

If you don't need automatic whitespace (or can do it manually for the rule), you can achieve a transitively silent rule by writing `rule = _{ atomic_rule }` and `atomic_rule = @{ actual ~ rule }`.

This isn't the best workaround, as it requires making the inner rules not eat implicit trivia, but it works.
